README.md

```diff
@@ -394,10 +396,57 @@ The `extract` method takes no arguments:
     1. `v` must have the same type that `f` returns in `extend`.
 
 
+### Bifunctor
 
+A value that implements the Bifunctor specification is a functor with two
+type arguments that are both covariant.
 
+1. `p.bimap(a => a, b => b)` is equivalent to `p` (identity)
+2. `p.bimap(compose(f1)(f2), compose(g1)(g2))` is equivalent to `p.bimap(f1, g1).bimap(f2, g2)` (composition)
 
+#### `bimap` method
 
+A value which has a Bifunctor must provide an `bimap` method. The `bimap`
+method takes two arguments:
+
+    c.bimap(f, g)
+
+1. `f` must be a function which returns a value
+
+    1. If `f` is not a function, the behaviour of `bimap` is unspecified.
```

Comment:
We need here "`f` can return any value." and "`bimap` must return a value of the same Bifunctor	" below for consistency with Functor.